aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,8 +1,315 @@
+				asp-fallback-test-class="carousel-caption" asp-fallback-test-property="display" asp-fallback-test-value="none" />
+		<link rel="stylesheet" href="~/css/site.css" asp-file-version="true" />
+	</environment>
+	
+Switching Between Environments
+------------------------------
+
+To switch between compiling for different environments, you can change the ASPNET_ENV environment variable. 
+1.	In **Task Runner Explorer**, verify that the **min** task has been set to occur **Before Build**.
+2.	In **Solution Explorer**, right-click the project name and select **Properties**.
+	The property sheet for the Web app is displayed.
+3.	Set the value of the **ASPNET_ENV** environment variable to ``Production``.
+4.	Press **F5** to run the application in a browser.
+5.	In the browser window, right-click the page and select **View Source** to see the html for the page.
+	You will notice that the stylesheet links point to the minified CSS files.
+6.	Close the browser to stop the Web app.
+7.	In Visual Studio, return to the property sheet for the Web app and change the **ASPNET_ENV** environment variable back to ``Development``.
+8.	Press **F5** to run the application in a browser again.
+9.	In the browser window, right-click the page and select **View Source** to see the html for the page.
+	You will notice that the stylesheet links point to the full version of the CSS files.
+	
+	For more information related to Visual Studio 2015 environments, see `Working with Multiple Environments <http://docs.asp.net/en/latest/fundamentals/environments.html>`_.
+	
+Task and Module Details
+-----------------------
+A gulp task is registered with a function name.  You can specify dependencies if other tasks must run before the current task. Additional functions allow you to run and watch the gulp tasks, as well as set the source (src) and destination (dest) of the files that you are modifying. The following are the primary gulp functions:
+
+•	gulp.task(name[, deps], fn) { }
+•	gulp.run(tasks) { }
+•	gulp.watch(glob [, opts], tasks) { }
+•	gulp.src(globs[, options]) { }
+•	gulp.dest(path[, options]) { } 
+
+For additional gulp API reference information, see `Gulp Docs API <https://github.com/gulpjs/gulp/blob/master/docs/API.md>`_. 
+
+Gulp Recipes
+------------
+The gulp community provides gulp `recipes <https://github.com/gulpjs/gulp/blob/master/docs/recipes/README.md>`_. These recipes are common scenarios to accomplish gulp tasks. 
+
+Summary
+-------
+Bundling and minification are two techniques you can use in ASP.NET 5 to improve performance. Gulp is a JavaScript-based streaming build toolkit that can be used for bundling and minification. Visual Studio 2015 automatically installs gulp along with a set of gulp tasks. Gulp is maintained on `GitHub <https://github.com/gulpjs/gulp>`_. For additional information about gulp, see the `Gulp Documentation <https://github.com/gulpjs/gulp/blob/master/docs/README.md>`_ on GitHub.
+
+
```

Comment:
check if this is still true
Bundles should be partitioned by pages that need them. For example, the default ASP.NET MVC template for an internet application creates a jQuery Validation bundle separate from jQuery. Because the default views created have no input and do not post values, they don't include the validation bundle.